### DISCUSSION TITLE
How to increase voting security by checking voter data

### DISCUSSION THREAD

**Discussion by InterestingIndeed** [OpenSlides/OpenSlides#5911] [Q&A]
We have been successfully using the voting tools implemented in openslides. I would love to increase the trust of the users by promoting voting supervisors, that check the voting process. When we are holding our assembly offline, voting supervisors are elected to watch the voting process, count votes etc.

I would love to have the option to export the voter data from openslides and make that data available for such a supervisor team. Additinally, I want to "prove" that I can erase the voter data after an essembly to ensure dataprotection. I am pretty new to hosting a server and finding data in the structure, openslides is my first big project an I am learning a lot. Where can I find the voter data OR where can I find a documentation of the voter tool that shows where the data is stored?

Best regards

**Comment by ostcar** [comment]
I do not understand what you are requesting.

What exactly should the voting supervisor? On a nominal vote, all data is publish. Everyone, even a voting supervisor, can check it. On a non-nominal vote, there is no way for anyone to get more data then it is shown on the client.

What exactly do you want to erase after an assembly? Do you want to remove the data on nominal votes or on non-nominal votes?

The vote data is stored in the database. [https://github.com/OpenSlides/OpenSlides/blob/master/server/openslides/poll/models.py](Here) is the code that generates the tables.

**Reply by InterestingIndeed** [reply]
I am trying to figure out how to move the voting process out of the blackbox. 

1. From my current experience, a lot of users are insecure about online voting, because they dont know whats going on in the background. Thank you for the link to the models!

2. People are insecure about what happens to their data and if someone somehow can check how they voted. The job of a supervisor would be to check the vote data for obvious errors in the case of non-nominal votes (e.g. "more votes than their should be", "irregular number of votes from one user").

3. In our case, voting ballots are archieved for a couple of years. The same thing could work, if I could extract the voter data. If it is stored in the database, I might be able to find it (that is more of an issue on my side, because I am very inexperienced and has nothing to do with openslides I guess). Thanks for telling me where I can find it.

4. Showing users where the data is stored and that you are able to delete or archieve the data might increase their trust towards openslides. I enjoy working with openslides and I trust the people who are working on this opensource solution. This is also my main argument when it comes to data security/voting security: "Dear assembly, because openslides is opensource, a lot of people check the code for errors, if there would be any errors, somebody would be reporting about it!" It is comparable to offline voting: "Here are the ballots, if you want to do a recount, feel free to do so!" 

Thank you for pointing me into the right directions, I will start looking into the database.

**Reply by normanjaeckel** [reply]
Please aware that using open source software like OpenSlides does not guarantee that the admin in your current assembly runs this code. If you take part in an assembly you can never check which code really runs. You only see some requests going out to the server and you see that you get some responses with the election result. And you can compare your results with some other participants. But you can not check what is going on on the server. Only the admin can do this and he can add extra code that extracts and or changes any voting data you have at any time. He can produce any result he likes in non-nominal votings. Only nominal votings are save here (and only if a respective part of participants checks the protocol afterwards). Thats why we currently do not promote OpenSlides for governmental elections (as we would not promote any other centralized electronic election system).
